prefect_gcp/cloud_run_job.py

```diff
@@ -0,0 +1,578 @@
+        ):
+            return True
+        return False
+
+    def is_ready(self) -> bool:
+        """Whether a job is finished registering and ready to be executed"""
+        if self._is_missing_container():
+            raise Exception(f"{self.ready_condition['message']}")
+        return self.ready_condition.get("status") == "True"
+
+    def has_execution_in_progress(self) -> bool:
+        """See if job has a run in progress."""
+        return (
+            self.execution_status == {}
+            or self.execution_status.get("completionTimestamp") is None
+        )
+
+    @staticmethod
+    def _get_ready_condition(job: dict) -> dict:
+        if job["status"].get("conditions"):
+            for condition in job["status"]["conditions"]:
+                if condition["type"] == "Ready":
+                    return condition
+
+        return {}
+
+    @staticmethod
+    def _get_execution_status(job: dict):
+        if job["status"].get("latestCreatedExecution"):
+            return job["status"]["latestCreatedExecution"]
+
+        return {}
+
+    @classmethod
+    def from_json(cls, job: dict):
+        """Construct a Job instance from a Jobs JSON response."""
+
+        return cls(
+            metadata=job["metadata"],
+            spec=job["spec"],
+            status=job["status"],
+            name=job["metadata"]["name"],
+            ready_condition=cls._get_ready_condition(job),
+            execution_status=cls._get_execution_status(job),
+        )
+
+
+class Execution(BaseModel):
+    name: str
+    metadata: dict
+    spec: dict
+    status: dict
+    log_uri: str
+
+    def is_running(self) -> bool:
+        """Returns True if Execution is not completed."""
+        return self.status.get("completionTime") is None
+
+    def condition_after_completion(self):
+        """Returns Execution condition if Execution has completed."""
+        for condition in self.status["conditions"]:
+            if condition["type"] == "Completed":
+                return condition
+
+    def succeeded(self):
+        """Whether or not the Execution completed is a successful state."""
+        completed_condition = self.condition_after_completion()
+        if completed_condition and completed_condition["status"] == "True":
+            return True
+
+        return False
+
+    @classmethod
+    def from_json(cls, execution):
+        """Create an Execution instance from a GCP Execution API request."""
+        return cls(
+            name=execution["metadata"]["name"],
+            metadata=execution["metadata"],
+            spec=execution["spec"],
+            status=execution["status"],
+            log_uri=execution["status"]["logUri"],
+        )
+
+
+class CloudRunJob(Infrastructure):
+    """Infrastructure block used to run GCP Cloud Run Jobs.
+
+    Project name information is provided by the Credentials object, and should always
+    be correct as long as the Credentials object is for the correct project.
+    """
```

Comment:
Should there be docstrings of fields?

Reply:
`mkdocstrings` should display the pydantic fields for us.